misk-crypto/src/test/kotlin/misk/crypto/PgpKeyTest.kt

```diff
@@ -0,0 +1,92 @@
+package misk.crypto
+
+import misk.config.MiskConfig
+import misk.crypto.pgp.PgpDecrypter
+import misk.crypto.pgp.PgpDecrypterProvider
+import misk.crypto.pgp.PgpEncrypter
+import misk.crypto.pgp.PgpEncrypterProvider
+import misk.resources.ResourceLoader
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import javax.inject.Inject
+
+@MiskTest(startService = true)
+class PgpKeyTest {
+  @Suppress("unused")
+  @MiskTestModule
+  val module = CryptoTestModule()
+
+  @Inject private lateinit var resourceLoader: ResourceLoader
+
+  @Test fun `encrypt and decrypt byte array armored`() {
+    val secretMessage = "foo bar baz here's a very secret message".toByteArray()
+
+    val pgpEncrypter = getPgpEncrypter(publicKeyPath, true)
+    val pgpDecrypter = getPgpDecrypter(privateKeyJsonPath, true)
+
+    val encryptedBytes = pgpEncrypter.encrypt(secretMessage, true)
+    val decryptedBytes = pgpDecrypter.decrypt(encryptedBytes)
+
+    assertThat(encryptedBytes).isNotEqualTo(secretMessage)
+    assertThat(decryptedBytes).isEqualTo(secretMessage)
+  }
+
+  @Test fun `encrypt and byte array stream`() {
+    val secretMessage = "foo bar baz here's a very secret message".toByteArray()
+
+    val pgpEncrypter = getPgpEncrypter(publicKeyPath, true)
+    val pgpDecrypter = getPgpDecrypter(privateKeyJsonPath, true)
+
+    val encryptedBytes = pgpEncrypter.encrypt(secretMessage, false)
+    val decryptedBytes = pgpDecrypter.decrypt(encryptedBytes)
+
+    assertThat(encryptedBytes).isNotEqualTo(secretMessage)
+    assertThat(decryptedBytes).isEqualTo(secretMessage)
+  }
+
+  @Test fun `fake pgp encrypter-decrypter`() {
+    val secretMessage = "foo bar baz here's a very secret message"
+    val secretMessageBase64 = "Zm9vIGJhciBiYXogaGVyZSdzIGEgdmVyeSBzZWNyZXQgbWVzc2FnZQ=="
+
+    val pgpEncrypter = getPgpEncrypter(publicKeyPath, false)
+    val pgpDecrypter = getPgpDecrypter(privateKeyJsonPath, false)
+
+    // The fake pgp encrypter/decrypter only uses base64.
+    val encryptedBytes = pgpEncrypter.encrypt(secretMessage.toByteArray(), true)
+    assertThat(encryptedBytes.decodeToString()).isEqualTo(secretMessageBase64)
+
+    val decryptedBytes = pgpDecrypter.decrypt(encryptedBytes)
+    assertThat(decryptedBytes.decodeToString()).isEqualTo(secretMessage)
+  }
+
+  private fun getPgpDecrypter(filePath: String, real: Boolean): PgpDecrypter {
```

Comment:
Checkout the MiskTestModule.

When you initialize the keys like this, you don't actually use the service modules to test this.
Instead, the keys should be provided and loaded via the configuration, so you'll actually test the entire flow (loading the key from the configuration, "decrypting" it, getting it from the provider and using it)

Reply:
well there's no tests around that anywhere in misk-crypto as far as i can tell? so i don't know why we need to add them specifically for this pr

Reply:
i guess there's marginally more testing with ```private fun getInjector(keyMap: List<Pair<String, KeysetHandle>>): Injector {
    val keys = keyMap.map {
      var keyType = KeyType.AEAD
      val keyTypeUrl = it.second.keysetInfo.getKeyInfo(0).typeUrl
      if (keyTypeUrl.endsWith("HmacKey")) {
        keyType = KeyType.MAC
      } else if (keyTypeUrl.endsWith("Ed25519PrivateKey")) {
        keyType = KeyType.DIGITAL_SIGNATURE
      } else if (keyTypeUrl.endsWith("EciesAeadHkdfPrivateKey")) {
        keyType = KeyType.HYBRID_ENCRYPT_DECRYPT
      } else if (keyTypeUrl.endsWith("EciesAeadHkdfPublicKey")) {
        keyType = KeyType.HYBRID_ENCRYPT
      } else if (keyTypeUrl.endsWith("AesSivKey")) {
        keyType = KeyType.DAEAD
      } else if (keyTypeUrl.endsWith("AesGcmHkdfStreamingKey")) {
        keyType = KeyType.STREAMING_AEAD
      }
      Key(it.first, keyType, generateEncryptedKey(it.second))
    }
    val config = CryptoConfig(keys, "test_master_key")
    return Guice.createInjector(CryptoTestModule(), CryptoModule(config), DeploymentModule.forTesting())
  }``` but like....not really